src/mon/OSDMonitor.cc

```diff
@@ -4172,6 +4194,20 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
       f->close_section();
     }
     f->flush(rdata);
+  } else if (prefix == "osd versions") {
+    if (!f)
```

Comment:
this is not necessary, `f` is always valid. see https://github.com/ceph/ceph/pull/15436/files#diff-0a5db46a44ae9900e226289a810f10e8R3975
```c++
  boost::scoped_ptr<Formatter> f(Formatter::create(format));
```